flytekit/core/tracker.py

```diff
@@ -273,6 +273,11 @@ def _resolve_abs_module_name(self, path: str, package_root: typing.Optional[str]
         if dirname == package_root:
             return basename
 
+        # Execution in a Jupyter notebook, we cannot resolve the module path
+        if not os.path.exists(dirname):
+            logger.warning(f"Directory {dirname} does not exist. It is likely that we are in a Jupyter notebook.")
```

Comment:
@thomasjpfan There are two possible cases that `dirname` doesn't exist in the current os. The first one is already mentioned. 
The second one is when we pickled the actual task in `map_task`, the `ArrayNodeMapTask` will try to get the module name [here](https://github.com/flyteorg/flytekit/blob/98d722fec491ecd68ba3ea92dfae9f676d32f677/flytekit/core/array_node_map_task.py#L101-L110) in the remote environment. In this case, since the entity is already pickled, it is not in the Jupyter Notebook environment. I can check whether the special file `PICKLE_FILE_PATH` exists in the current os for the check. WDYT